docs/core/extensions/artificial-intelligence.md

```diff
@@ -0,0 +1,650 @@
+
+- <xref:Microsoft.Extensions.AI.AudioContent>
+- <xref:Microsoft.Extensions.AI.DataContent>
+- <xref:Microsoft.Extensions.AI.FunctionCallContent>
+- <xref:Microsoft.Extensions.AI.FunctionResultContent>
+- <xref:Microsoft.Extensions.AI.ImageContent>
+- <xref:Microsoft.Extensions.AI.TextContent>
+- <xref:Microsoft.Extensions.AI.UsageContent>
+
+#### Request chat completion with streaming
+
+The inputs to <xref:Microsoft.Extensions.AI.IChatClient.CompleteStreamingAsync*?displayProperty=nameWithType> are identical to those of `CompleteAsync`. However, rather than returning the complete response as part of a <xref:Microsoft.Extensions.AI.ChatCompletion> object, the method returns an <xref:System.Collections.Generic.IAsyncEnumerable`1> where `T` is <xref:Microsoft.Extensions.AI.StreamingChatCompletionUpdate>, providing a stream of updates that collectively form the single response.
+
+```csharp
+using Microsoft.Extensions.AI;
+
+IChatClient client = new SampleChatClient(
+    new Uri("http://coolsite.ai"), "my-custom-model");
+
+await foreach (var update in client.CompleteStreamingAsync("What is AI?"))
+{
+    Console.Write(update);
+}
+```
+
+> [!TIP]
+> Streaming APIs are nearly synonymous with AI user experiences. C# enables compelling scenarios with its `IAsyncEnumerable<T>` support, allowing for a natural and efficient way to stream data.
+
+#### Tool calling
+
+Some models and services support _tool calling_, where requests can include tools for the model to invoke functions to gather additional information. Instead of sending a final response, the model requests a function invocation with specific arguments. The client then invokes the function and sends the results back to the model along with the conversation history. The `Microsoft.Extensions.AI` library includes abstractions for various message content types, including function call requests and results. While consumers can interact with this content directly, `Microsoft.Extensions.AI` automates these interactions and provides:
+
+- <xref:Microsoft.Extensions.AI.AIFunction>: Represents a function that can be described to an AI service and invoked.
+- <xref:Microsoft.Extensions.AI.AIFunctionFactory>: Provides factory methods for creating commonly used implementations of `AIFunction`.
+- <xref:Microsoft.Extensions.AI.FunctionInvokingChatClient>: Wraps an `IChatClient` to add automatic function invocation capabilities.
+
+Consider the following example that demonstrates a random function invocation:
+
+```csharp
+using System.ComponentModel;
+using Microsoft.Extensions.AI;
+
+[Description("Gets the current weather")]
+string GetCurrentWeather() => Random.Shared.NextDouble() > 0.5
+    ? "It's sunny"
+    : "It's raining";
+
+IChatClient client = new ChatClientBuilder(
+        new OllamaChatClient(new Uri("http://localhost:11434"), "llama3.1"))
```

Comment:
Potential change of switching from llama3 in these samples to phi3 since phi is a Microsoft developed model.

Reply:
I'm fine with whatever. If you think it's worth changing, happy to do so. Sometimes it's nice to show that we play nicely in supporting non-MS things too, as a logical argument for leaving this is as is. But again, I'm fine either way.